back/src/database/schema.prisma

```diff
@@ -8,25 +8,110 @@ datasource db {
+
+    // n-m self relation : an user can be have friends and an user can be friends of other users.
+    friends   User[] @relation("UserFriends")
+    friendsOf User[] @relation("UserFriends")
+
+    matchAsUserOne Match[] @relation("UserOne")
+    matchAsUserTwo Match[] @relation("UserTwo")
+    matchesWon     Match[] @relation("WinnerUser")
+    matchesLost    Match[] @relation("LooserUser")
+
+    // Here i define an explicit many-to-many relations.
+    // Three models defines this kind of relation :
+    //    User and UserChannel has a many-to-many relations (an User can be in multiple channel and a Channel can have multiple User)
+    //    UserOnChannel, that's the link or pivot table.
+    // This pivot table defines additional fields that describes the User / UserChannel relationship,
+    // such as the nickname, the status of the User on a particural Channel etc...
+    // See https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations
+
+    onChannels UserOnChannel[]
+}
+
+model UserOnChannel {
+    user      User                @relation(fields: [userId], references: [id])
+    userId    Int
+    joinedAt  DateTime            @default(now())
+    role      UserOnChannelRole   @default(USER)
+    status    UserOnChannelStatus @default(REGULAR)
+    channel   UserChannel         @relation(fields: [channelId], references: [id])
+    channelId Int
+
+    @@id([userId, channelId])
+}
+
+model UserChannel {
+    id           Int                   @id @unique @default(autoincrement())
+    createdAt    DateTime              @default(now())
+    visibility   UserChannelVisibility @default(PUBLIC)
+    name         String?
+    participants UserOnChannel[]
 }
 
-model Profile {
+model UserProfile {
     id            Int  @id @unique @default(autoincrement())
     playedMatches Int  @default(0)
     wonMatches    Int  @default(0)
     user          User @relation(fields: [userId], references: [id])
     userId        Int  @unique
 }
 
+model Match {
+    id         Int       @id @unique @default(autoincrement())
+    createdAt  DateTime  @default(now())
+    finishedAt DateTime?
+    bounces    Int
+    userOne    User      @relation("UserOne", fields: [userOneId], references: [id])
+    userTwo    User      @relation("UserTwo", fields: [userTwoId], references: [id])
+    winner     User      @relation("WinnerUser", fields: [winnerId], references: [id])
+    looser     User      @relation("LooserUser", fields: [looserId], references: [id])
+    userOneId  Int
+    userTwoId  Int
+    winnerId   Int
+    looserId   Int
+}
+
 model AuthIdentifier {
     identifier String @id @unique
     user       User   @relation(fields: [userId], references: [id])
     userId     Int
 }
+
+enum UserStatus {
+    ONLINE
+    OFFLINE
+    PLAYING
+}
+
+enum UserOnChannelRole {
+    OPERATOR
+    ADMIN
+    USER
+}
+
+enum UserOnChannelStatus {
+    REGULAR
+    MUTED
+    BANNED
+    KICKED
```

Comment:
Ca dépendra de l'implémentation mais pour moi `KICKED` n est pas un status, mais le player est juste retiré du channel

Reply:
True that